lib/hanami/events/adapter/redis.rb

```diff
@@ -43,11 +44,9 @@ def thread_spawned!
           @thread_spawned = true
         end
 
-        def subscribe_to_events(connection)
-        end
-
-        def call_listeners(event_name, message)
-          @listeners.each { |listener| listener.call(event_name, JSON.parse(message)) }
+        def call_subscribers(payload)
+          event_name = payload.delete('event_name')
+          @subscribers.each { |subscriber| subscriber.call(event_name, payload) }
```

Comment:
And also I think we can replace this line to the method if future because memory adapter has this method too. And (in future) create base adapter class. But now it's okay